packages/astro/src/vite-plugin-astro-server/request.ts

```diff
@@ -37,7 +37,7 @@ export async function handleRequest({
 	if (config.trailingSlash === 'never' && !incomingRequest.url) {
 		pathname = '';
 	} else {
-		pathname = decodeURI(url.pathname);
+		pathname = url.pathname;
```

Comment:
It doesn't look like we need to decode it here. It also doesn't work well with the line below:

https://github.com/withastro/astro/blob/cf993bc263b58502096f00d383266cd179f331af/packages/astro/src/vite-plugin-astro-server/request.ts#L43-L44

(URLs are always encoded)

Removing this `decodeURI` seems to not break things. It affects dev only. The prod SSR "app" also doesn't decode the url, so I think this makes it more consistent.